agenta-backend/agenta_backend/services/evaluation_service.py

```diff
@@ -917,3 +1042,67 @@ async def fetch_custom_evaluation_names(
             )
         )
     return list_of_custom_eval_names
+
+
+async def evaluate_in_bulk(new_evaluation: BulkEvaluationDB, **user_org_data: dict):
+    testset = await db_manager.fetch_testset_by_id(new_evaluation.testset.id)
+
+    for variant_id in new_evaluation.variants:
+        variant_id = str(variant_id)
+        app_variant_db = await db_manager.fetch_app_variant_by_id(variant_id)
+        deployment = await db_manager.get_deployment_by_objectid(
+            app_variant_db.base.deployment
+        )
+
+        uri = deployment.uri.replace("http://localhost", "http://host.docker.internal")
+
+        for input in testset.csvdata:
```

Comment:
this part should be a task in celery, no?

Reply:
we want to do everything in the task. the current approach does not work. we are outsourcing just the metric computation to celery, however what takes most of the time is generation. 
we need to outsource everything to celery:
- load test set
- run llm app
- call eval methods
- save results

Basically a task taking
( one llm app, one test set, list[evalMethods]) which does everything and saves the results./

Reply:
@aakrem -> this is most important comment in the review